src/java/org/apache/cassandra/db/compaction/BackgroundCompactionRunner.java

```diff
@@ -382,6 +385,7 @@ private CompletableFuture<Void> startTask(ColumnFamilyStore cfs, AbstractCompact
         {
             ongoingCompactions.decrementAndGet();
             logger.debug("Background compaction task for {} was rejected", cfs);
+            task.rejected(ex);
```

Comment:
Are we relying on this to call the observer with the error? I see a warning in my IDE because we're ignoring the resulting throwable. Do we want to return a failed future on the next line? It's minor, but a comment would make that clearer.

Reply:
Done